gen_ai/task2.1/Michael_Soffer_refactor_code.py

```diff
@@ -0,0 +1,119 @@
+import os
+import sys
+import logging
+import subprocess
+from dotenv import load_dotenv
+
+# Load environment variables
+load_dotenv("../../.env")
+load_dotenv("../.env")
+
+# === CONFIGURATION ===
+DB_USER = "root"
+DB_PASS = os.getenv("MARIADB_ROOT_PASSWORD")
+DB_NAME = os.getenv("MARIADB_DATABASE")
+
+RESERVED_IDS = list(range(1999900, 1999905))
+RESERVED_ITEM_IDS = list(range(200000, 200020))
+HARDCODED_LOCATIONS = [
+    (-72.26, -192.6, 15.09),
+    (-62.77, -193.64, 16.17),
+    (-51.89, -183.39, 14.4),
+    (-41.97, -178.06, 13.36),
+    (-72.72, -195.64, 34.2),
+]
+QUEST_ZONE = "tutorialb"
+
+# === LOGGING ===
+def setup_logging():
+    logging.basicConfig(
+        level=logging.INFO,
+        format="%(asctime)s [%(levelname)s] [%(name)s] %(message)s",
+    )
+    return logging.getLogger(os.path.basename(__file__))
```

Comment:
this is a much cleaner implementation of getting the file name for the logger.

along with having logging being setup in main() and more concise logging descriptions. its an overall improvement.